File: src/demos/ps/maps/Sopia.java
package demos.ps.maps;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import static demos.ps.oo.PSGame.*;
import demos.ps.oo.City;
import demos.ps.oo.PSGame;
import demos.ps.oo.Item;
import demos.ps.oo.PSGame.Flags;
import demos.ps.oo.PSLibItem.OriginalItem;
import demos.ps.oo.PSMenu;
import demos.ps.oo.PSMenu.EntityClothes;
import demos.ps.oo.PSMenu.EntityType;
import demos.ps.oo.PSMenu.Scene;
import demos.ps.oo.PSMenu.SpecialEntity;

public class Sopia {

	private static final Logger log = LogManager.getLogger(Sopia.class);

	public static void hospital() {
		PSMenu.startScene(Scene.HOSPITAL_VILLAGE, EntityType.VILLA_WMN_BROWN, EntityClothes.WHITE);
		PSGame.Hospital(2); // More expensive
		PSMenu.endScene();
	}
	
	public static void house1() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.VILLA_WMN_BROWN, EntityClothes.BLUE);
		PSMenu.Stext(getString("Sopia_House_1"));
		PSMenu.endScene();
	}

	public static void house2() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.VILLA_MAN_BLUE, EntityClothes.RED);
		PSMenu.Stext(getString("Sopia_House_2"));
		PSMenu.endScene();
	}
	public static void house3() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, SpecialEntity.OLDMAN);
		if(PSMenu.Prompt(getString("Sopia_House_3"), getYesNo()) == 1) {
			if(PSGame.getParty().mst < 400) {
				PSMenu.StextLast(getString("Sopia_House_3NotEnough"));
			}
			else {
				PSMenu.StextLast(getString("Sopia_House_3Yes"));
				PSGame.getParty().mst-=400;
				PSGame.setFlag(Flags.INFO_PERSEUS);
			}
		} else {
			PSMenu.StextLast(getString("Sopia_House_3No"));
		}
		PSMenu.endScene();
	}

	public static void house4() {
		PSMenu.startScene(Scene.RUINED_HOUSE, EntityType.VILLA_WMN_BROWN, EntityClothes.GREEN);
		if(PSMenu.Prompt(getString("Sopia_House_4"), getYesNo()) == 1) {
				PSMenu.StextLast(getString("Sopia_House_4Yes"));
		} else {
			PSMenu.StextLast(getString("Sopia_House_4No"));
		}
		PSMenu.endScene();
	}	
	
	
	public static void house5() {
		PSMenu.startScene(Scene.RUINED_HOUSE, SpecialEntity.BEGGAR);
		if(PSMenu.Prompt(getString("Sopia_House_5"), getYesNo()) == 1) {
			
			boolean foundCola = PSGame.findItemWithParty(OriginalItem.Inventory_Monomate, true);
			if(foundCola) {
				PSMenu.StextLast(getString("Sopia_House_5Yes"));				
			} else {
				PSMenu.StextLast(getString("Sopia_House_5NotEnough"));
			}
		} else {
			PSMenu.StextLast(getString("Sopia_House_5No"));	
		}

		PSMenu.endScene();
	}
	public static void house6() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.VILLA_MAN_BROWN, EntityClothes.BLUE);
		PSMenu.Stext(getString("Sopia_House_6"));
		PSMenu.endScene();
	}
	public static void house7() {
		PSMenu.startScene(Scene.RUINED_HOUSE, SpecialEntity.BEGGAR);
		if(PSMenu.Prompt(getString("Sopia_House_7"), getYesNo()) == 1) {
				PSMenu.StextLast(getString("Sopia_House_7Yes"));
		} else {
			PSMenu.StextLast(getString("Sopia_House_7No"));
		}
		PSMenu.endScene();
	}	
	
	

	
	public static void exit() {
		mapswitch(Planet.MOTAVIA, City.SOPIA.getX(), City.SOPIA.getY());
	}
}

File: src/demos/ps/maps/Sopia.ts
/**
 * Sopia Village Script
 * TypeScript port of Sopia.java
 */

import { PSGame } from '../PSGame';
import { Flags } from '../game/GameData';
import { Planet, City, CityHelper } from '../game/City';
import { OriginalItem } from '../game/PSLibItem';
import { PSSceneType, EntityType, EntityClothes, SpecialEntity, PSMenu } from '../PSMenu';

export class Sopia {

  public static async hospital(): Promise<void> {
    await PSMenu.startScene(PSSceneType.HOSPITAL_VILLAGE, EntityType.VILLA_WMN_BROWN, EntityClothes.WHITE);
    await PSGame.Hospital(2); // More expensive
    await PSMenu.endScene();
  }

  public static async house1(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.VILLA_WMN_BROWN, EntityClothes.BLUE);
    await PSMenu.Stext(PSGame.getString("Sopia_House_1"));
    await PSMenu.endScene();
  }

  public static async house2(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.VILLA_MAN_BLUE, EntityClothes.RED);
    await PSMenu.Stext(PSGame.getString("Sopia_House_2"));
    await PSMenu.endScene();
  }

  public static async house3(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, SpecialEntity.OLDMAN);
    if (await PSMenu.Prompt(PSGame.getString("Sopia_House_3"), PSGame.getYesNo()) === 1) {
      if (PSGame.getParty().mst < 400) {
        await PSMenu.StextLast(PSGame.getString("Sopia_House_3NotEnough"));
      } else {
        await PSMenu.StextLast(PSGame.getString("Sopia_House_3Yes"));
        PSGame.getParty().mst -= 400;
        PSGame.setFlag(Flags.INFO_PERSEUS);
      }
    } else {
      await PSMenu.StextLast(PSGame.getString("Sopia_House_3No"));
    }
    await PSMenu.endScene();
  }

  public static async house4(): Promise<void> {
    await PSMenu.startScene(PSSceneType.RUINED_HOUSE, EntityType.VILLA_WMN_BROWN, EntityClothes.GREEN);
    if (await PSMenu.Prompt(PSGame.getString("Sopia_House_4"), PSGame.getYesNo()) === 1) {
      await PSMenu.StextLast(PSGame.getString("Sopia_House_4Yes"));
    } else {
      await PSMenu.StextLast(PSGame.getString("Sopia_House_4No"));
    }
    await PSMenu.endScene();
  }

  public static async house5(): Promise<void> {
    await PSMenu.startScene(PSSceneType.RUINED_HOUSE, SpecialEntity.BEGGAR);
    if (await PSMenu.Prompt(PSGame.getString("Sopia_House_5"), PSGame.getYesNo()) === 1) {
      const foundCola = PSGame.findItemWithParty(OriginalItem.Inventory_Monomate, true);
      if (foundCola) {
        await PSMenu.StextLast(PSGame.getString("Sopia_House_5Yes"));
      } else {
        await PSMenu.StextLast(PSGame.getString("Sopia_House_5NotEnough"));
      }
    } else {
      await PSMenu.StextLast(PSGame.getString("Sopia_House_5No"));
    }
    await PSMenu.endScene();
  }

  public static async house6(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.VILLA_MAN_BROWN, EntityClothes.BLUE);
    await PSMenu.Stext(PSGame.getString("Sopia_House_6"));
    await PSMenu.endScene();
  }

  public static async house7(): Promise<void> {
    await PSMenu.startScene(PSSceneType.RUINED_HOUSE, SpecialEntity.BEGGAR);
    if (await PSMenu.Prompt(PSGame.getString("Sopia_House_7"), PSGame.getYesNo()) === 1) {
      await PSMenu.StextLast(PSGame.getString("Sopia_House_7Yes"));
    } else {
      await PSMenu.StextLast(PSGame.getString("Sopia_House_7No"));
    }
    await PSMenu.endScene();
  }

  public static exit(): void {
    PSGame.mapswitchToPlanet(Planet.MOTAVIA, CityHelper.getX(City.SOPIA), CityHelper.getY(City.SOPIA));
  }
}
